heat/sparse/factories.py

```diff
@@ -94,7 +94,7 @@ def sparse_csr_matrix(
     (indptr: tensor([0, 1, 3, 4]), indices: tensor([2, 0, 2, 2]), data: tensor([1, 1, 2, 3]), dtype=ht.int64, device=cpu:0, split=None)
     """
     # version check
-    if int(torch.__version__.split(".")[1]) < 10:
+    if torch.__version__ < "1.10":
```

Comment:
oh, I didn't know this was possible! 

I had to go and time these two methods. We do many many checks, and those millionths of seconds pile up. 🤭 

Turns out the string check (this PR) is about 10 times slower than the int check (previous solution) even after having to manually split the string. The int check is even faster for torch >=2, as the first condition is sufficient to make the decision.

Summary: I'd go for 
```python
if int(torch.__version__.split(".")[0]) <= 1 and int(torch.__version__.split(".")[1]) < 10
```
solution in `factories` as well. What do you think?

Reply:
Interesting to know that it is soo much slower.
I have reverted back to the older method of version checking.